core/chaincode/lifecycle/lifecycle.go

```diff
@@ -467,11 +466,10 @@ func (ef *ExternalFunctions) ApproveChaincodeDefinitionForOrg(chname, ccname str
 					if err := ef.Resources.Serializer.Deserialize(ChaincodeSourcesName, privateName, metadata, ccLocalPackage, orgState); err != nil {
 						return errors.WithMessagef(err, "could not deserialize chaincode package for %s", privateName)
 					}
-					uncommittedPackageID = ccLocalPackage.PackageID
-				}
 
-				if uncommittedPackageID == packageID {
-					return errors.Errorf("attempted to redefine uncommitted sequence (%d) for namespace %s with unchanged content", requestedSequence, ccname)
+					if ccLocalPackage.PackageID == packageID {
+						return errors.Errorf("attempted to redefine uncommitted sequence (%d) for namespace %s with unchanged content", requestedSequence, ccname)
+					}
```

Comment:
what if the `ok` at line 464 is false? we would still fail the check before this change, no?

Reply:
Not sure I'm following you. You're referring to the error directly above? 

Unless I'm missing something here, the only place we should care about checking the package id at this point in the code is if there's already been an approval for an uncommitted definition. If we receive a packageID here that's empty, we have problems elsewhere.

Reply:
will there ever be a case where ChaincodeSource metadata is empty (so the `ok` at line 464 is `false`), and we are passing in a non-empty pkgID? In this case, we used to error out, but now proceed. I might likely be misinterpreting this code though...

Reply:
I think this PR doesn't change the logic - only avoids a local variable declaration (which probably the compiler simplifies anyway). I'll go ahead and merge, if you feel there's an issue you can open a JIRA or revert